File: C#/Longest_Common_Subsequence_algorithm.cs
using System;

class LongestCommonSubsequence
{
    public static string FindLCS(string sequence1, string sequence2)
    {
        int[,] lengths = new int[sequence1.Length + 1, sequence2.Length + 1];

        for (int i = 0; i <= sequence1.Length; i++)
        {
            for (int j = 0; j <= sequence2.Length; j++)
            {
                if (i == 0 || j == 0)
                {
                    lengths[i, j] = 0;
                }
                else if (sequence1[i - 1] == sequence2[j - 1])
                {
                    lengths[i, j] = lengths[i - 1, j - 1] + 1;
                }
                else
                {
                    lengths[i, j] = Math.Max(lengths[i - 1, j], lengths[i, j - 1]);
                }
            }
        }

        int index = lengths[sequence1.Length, sequence2.Length];
        char[] lcs = new char[index];

        int m = sequence1.Length, n = sequence2.Length;
        while (m > 0 && n > 0)
        {
            if (sequence1[m - 1] == sequence2[n - 1])
            {
                lcs[index - 1] = sequence1[m - 1];
                m--;
                n--;
                index--;
            }
            else if (lengths[m - 1, n] > lengths[m, n - 1])
            {
                m--;
            }
            else
            {
                n--;
            }
        }

        return new string(lcs);
    }

    static void Main(string[] args)
    {
        string sequence1 = "AGGTAB";
        string sequence2 = "GXTXAYB";

        string lcs = FindLCS(sequence1, sequence2);
        Console.WriteLine("Longest Common Subsequence: " + lcs);
    }
}

File: JavaScripts/longest common subsequence_algorithm.js
function longestCommonSubsequence(str1, str2) {
    // Create a 2D array to store the lengths of LCS for each pair of substrings
    const dp = [];
    for (let i = 0; i <= str1.length; i++) {
        dp[i] = new Array(str2.length + 1).fill(0);
    }

    // Fill the dp array using dynamic programming approach
    for (let i = 1; i <= str1.length; i++) {
        for (let j = 1; j <= str2.length; j++) {
            if (str1[i - 1] === str2[j - 1]) {
                dp[i][j] = dp[i - 1][j - 1] + 1; // If characters match, increment LCS length
            } else {
                dp[i][j] = Math.max(dp[i - 1][j], dp[i][j - 1]); // Otherwise, take maximum of previous lengths
            }
        }
    }

    // Reconstruct LCS from dp array
    let lcs = '';
    let i = str1.length, j = str2.length;
    while (i > 0 && j > 0) {
        if (str1[i - 1] === str2[j - 1]) {
            lcs = str1[i - 1] + lcs; // Add matched character to LCS
            i--;
            j--;
        } else if (dp[i - 1][j] > dp[i][j - 1]) {
            i--; // Move to previous row if left cell has higher value
        } else {
            j--; // Move to previous column if top cell has higher value
        }
    }

    return lcs;
}

// Example strings
const str1 = 'ABCBDAB';
const str2 = 'BDCAB';

// Find the longest common subsequence
const result = longestCommonSubsequence(str1, str2);
console.log('Longest Common Subsequence:', result);
